prediction_prophet/autonolas/research.py

```diff
@@ -1177,7 +1186,8 @@ def make_prediction(
     additional_information: str,
     temperature: float = 0.7,
     engine: str = "gpt-3.5-turbo-0125",
-    api_key: SecretStr | None = None
+    api_key: SecretStr | None = None,
+    add_langfuse_callback: bool = False,
```

Comment:
This PR isn't so interesting, just adding callbacks everywhere and changing return types somewhere to contain more information.